diff --git a/src/base/net/stratum/Url.cpp b/src/base/net/stratum/Url.cpp
index 7904a445..d37adf08 100644
--- a/src/base/net/stratum/Url.cpp
+++ b/src/base/net/stratum/Url.cpp
@@ -33,6 +33,7 @@
 namespace xmrig {
 
 static const char kStratumTcp[]            = "stratum+tcp://";
+static const char kStratumTls[]            = "stratum+tls://";
 static const char kStratumSsl[]            = "stratum+ssl://";
 static const char kSOCKS5[]                = "socks5://";
 
@@ -86,6 +87,10 @@ bool xmrig::Url::parse(const char *url)
             m_scheme = STRATUM;
             m_tls    = false;
         }
+        else if (strncasecmp(url, kStratumTls, sizeof(kStratumTls) - 1) == 0) {
+            m_scheme = STRATUM;
+            m_tls    = true;
+        }
         else if (strncasecmp(url, kStratumSsl, sizeof(kStratumSsl) - 1) == 0) {
             m_scheme = STRATUM;
             m_tls    = true;
